src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -748,6 +764,19 @@ private void CancelOpenAndWait()
         public override Task OpenAsync(CancellationToken cancellationToken)
         {
             PrepareStatisticsForNewConnection();
+            
+            if (StatisticsEnabled ||
+                _diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
+            {
+                if (null == _statistics)
```

Comment:
@saurabh500 I have refactored this logic into `PrepareStatisticsForNewConnect()` to eliminate the redundancy.  These changes can be found in commit https://github.com/dotnet/corefx/pull/8420/commits/319736ba1954415602e7aecf25ba52daee275f36 Thanks! :smile: